python/tests/test_coaltime_distribution.py

```diff
@@ -952,42 +989,117 @@ def test_trees_per_block(self):
         assert np.allclose(tpb, 2)
 
 
+class TestCoalescenceTimeDistributionBlockedVsUnblocked(
+    TestCoalescenceTimeDistribution
+):
+    """
+    Test that methods give the same result regardless of how trees are blocked.
+    """
+
+    def coalescence_time_distribution(self, num_blocks=1):
+        ts = self.ts_many_edge_diffs()
+        sample_sets = [list(range(10)), list(range(20, 40)), list(range(70, 80))]
+        distr = ts.coalescence_time_distribution(
+            sample_sets=sample_sets,
+            weight_func="pair_coalescence_events",
+            blocks_per_window=num_blocks,
+            span_normalise=True,
+        )
+        return distr
+
+    def test_ecdf(self):
+        distr_noblock = self.coalescence_time_distribution(num_blocks=1)
+        distr_block = self.coalescence_time_distribution(num_blocks=10)
+        t = np.linspace(0, distr_noblock.tables[0].time[-1] + 1, 5)
+        assert np.allclose(distr_noblock.ecdf(t), distr_block.ecdf(t))
+
+    def test_num_coalesced(self):
+        distr_noblock = self.coalescence_time_distribution(num_blocks=1)
+        distr_block = self.coalescence_time_distribution(num_blocks=10)
+        t = np.linspace(0, distr_noblock.tables[0].time[-1] + 1, 5)
+        assert np.allclose(distr_noblock.num_coalesced(t), distr_block.num_coalesced(t))
```

Comment:
Using ``np.testing.assert_allclose(..)`` is a bit more helpful as it'll give some info on what's different. should be an easy search and replace on this file (since it's new)